### PR TITLE
fix(superuser): added single-tenant flag to validate password and u2f and not check superuser access form

### PR DESCRIPTION
**Pull request by maxiuyuan**
Currently for ST, the superuser modal is not able to verify the user's password or u2f. Using `VALIDATE_SUPERUSER_ACCESS_CATEGORY_AND_REASON` to allow password/u2f validation. In addition, The flag will disable the check for the superuser access form in the superuser class. The setting will be added to the ConfigStore so the FE can also check the Flag

FE: https://github.com/getsentry/sentry/pull/34670

### CODE REVIEW

File: src/sentry/api/endpoints/auth_index.py [hunk 1]
```diff
@@ -99,14 +99,15 @@ def _validate_superuser(self, validator, request):
...
         )
+        IS_SINGLE_TENANT = getattr(settings, "SENTRY_SINGLE_TENANT", False)
 
         # TODO Look at AuthVerifyValidator
         validator.is_valid()
         authenticated = None
 
         def _require_password_or_u2f_check():
-            if not is_self_hosted():
-                # Don't need to check as its only for self-hosted users
```
Comment (dashed) [inline src/sentry/api/endpoints/auth_index.py:109]: @maxiuyuan is it possible to add a test for this? 👀

**Review by rgibert — COMMENTED** [review]
Looks good here, I'll add SENTRY_SINGLE_TENANT=True to the ST setup for the next deploy.

**Review by dashed — APPROVED** [review]

**Comment by maxiuyuan** [comment]
> Looks good here, I'll add SENTRY_SINGLE_TENANT=True to the ST setup for the next deploy.

I'll merge this tomorrow. As sentry shouldn't be aware of ST, setting this `VALIDATE_SUPERUSER_ACCESS_CATEGORY_AND_REASON = False` for ST will fix the problem once merged

**Comment by rgibert** [comment]
Thanks for the quick turnaround on this @maxiuyuan

**Pull request merged**